prometheus_client/registry.py

```diff
@@ -135,10 +135,20 @@ def __init__(self, names, registry):
         self._registry = registry
 
     def collect(self):
-        for metric in self._registry.collect():
-            m = metric._restricted_metric(self._name_set)
-            if m:
-                yield m
+        names = copy.copy(self._name_set)
```

Comment:
Why is this copy here? This set shouldn't be changing after the object is instantiated.

Reply:
Trying to do it in the same way collections used to be filtered with target_info. There used to be a set variable that was modified if target_info was found - in order not to change the class field in the same way, we copy it to `names`.